### ISSUE TITLE
Use `<article>`s for cards when appropriate

### ISSUE THREAD

**Issue by MartijnCuppens** [twbs/bootstrap#30266]
The cards should be articles in our docs. We should have a look at this after https://github.com/twbs/bootstrap/pull/28517 is merged.

**Comment by rubenRP** [comment]
The tests have failed because there is a rule that requires adding headings in all blocks of type <article> (Consider using "h2" - "h6" elements to add identifying headings to all articles.), but I'm not sure that an element of type "Card" must contain them.

**Comment by MartijnCuppens** [comment]
thanks for looking into this, @rubenRP. I think we should only use `<article>`s for card with titles. And maybe also add a note to the docs that `<article>` should only be used if the card contains a title.

**Comment by XhmikosR** [comment]
Why should the cards be articles?

**Comment by MartijnCuppens** [comment]
> Why should the cards be articles?

It's just a semantic improvement. See https://html.spec.whatwg.org/multipage/sections.html#the-article-element

**Comment by XhmikosR** [comment]
@patrickhlauke can you chime in please?

I personally don't think we can just replace all of our docs cards with articles, hence the validator errors.

**Comment by MartijnCuppens** [comment]
> I personally don't think we can just replace all of our docs cards with articles, hence the validator errors.

We can't replace _all_ cards and shouldn't ignore these errors indeed.

**Comment by patrickhlauke** [comment]
personally, i'm not sure i see a strong enough need for using `<article>` in general, unless i was doing something like a list of news items or similar. would developers cargo-cult this for other uses, like notes/panels and that sort of thing? if so, they'll end up overcomplicating their structure for no reason (and making their pages overly verbose for AT users, who'd constantly hear when entering/exiting an article section).

**Comment by XhmikosR** [comment]
Thanks @patrickhlauke for the insights. That's what I was also afraid of.

**Comment by patrickhlauke** [comment]
a slightly more neutral alternative would also be `<section>` https://html.spec.whatwg.org/multipage/sections.html#the-section-element which doesn't have that implication that `<article>` has about being self-contained and being potentially ripped out of the page as a stand-alone complete piece of content.

but otherwise, perhaps keep the markup as `<div>` elements, but add a note saying that of course these classes can also be applied to other containers, like `<section>`, `<article>`, if the authors feel their content requires clearer/stricter structuring (and just in terms of how AT handle stuff, @stevefaulkner reminded me that if a `<section>` doesn't have a heading, it will be exposed in exactly the same way as a `<div>`)

**Comment by masi** [comment]
If article or any other ARIA regions makes sense for a card depends on the context. I have used the card deign pattern with article in the past when I saw fir, but that hasn't been always the case.

**Comment by masi** [comment]
Please note also my suggestion in issue #32555.

**Comment by mdo** [comment]
Lumping in with #32555.
